lib/components/joystick/joystick_action.dart

```diff
@@ -153,16 +154,16 @@ class JoystickAction {
 
       // Distance between the center of joystick background & drag position
       final centerPosition =
-          Position.fromOffset(_rectBackgroundDirection.center);
-      final dragPosition = Position.fromOffset(_dragPosition);
-      double dist = centerPosition.distance(dragPosition);
+          Vector2Factory.fromOffset(_rectBackgroundDirection.center);
```

Comment:
I think these can be extension functions on offset.

so you can do `_rectBackgroundDirection.center.toVector2()`

Reply:
similarly to the others

Reply:
Good idea!